### PR TITLE
Triplanar mapping node

### PR DESCRIPTION
**Pull request by Alpistinho**
I've added support for the triplanar mapping (https://github.com/LuxCoreRender/LuxCore/pull/252)
There is also some fixes for issues that I've found with my last pull request

image

### CODE REVIEW

**Comment by Alpistinho** [comment]
That is the result from this node tree
rock

File: nodes/materials/tree.py [hunk 1]
```diff
@@ -75,6 +75,7 @@ def poll(cls, context):
     LuxCoreNodeCategoryMaterial("LUXCORE_MATERIAL_TEXTURE", "Texture", items=[
         NodeItemMultiImageImport(),
         NodeItem("LuxCoreNodeTexImagemap", label="Image"),
+        NodeItem("LuxCoreNodeTriplanar", label="Triplanar mapping"),
```
Comment (Alpistinho) [inline nodes/materials/tree.py]: Oops, some mismatched capitalization here

File: export/aovs.py [hunk 2]
```diff
@@ -132,6 +132,9 @@ def convert(exporter, scene, context=None, engine=None):
         props = utils.create_props(prefix, definitions)
         props.Set(pipeline_props)
 
+        if (use_adaptive_sampling and final):
+            props.Set(pyluxcore.Property("film.noiseestimation.index", adaptive_sampling_film_pipeline_index))
+
```
Comment (Theverat) [inline export/aovs.py]: I would undo the changes in this file, my fixes were enough in my opinion.
I moved this code to line 130.

File: nodes/textures/triplanar.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+
+class LuxCoreNodeTriplanar(bpy.types.Node, LuxCoreNodeTexture):
+    bl_label = "Triplanar mapping"
+    bl_width_default = 160
+
+    def init(self, context):
+        self.add_input("LuxCoreSocketColor", "Texture 1", [0.8, 0.0, 0.0])
+        self.add_input("LuxCoreSocketColor", "Texture 2", [0.0, 0.8, 0.0])
+        self.add_input("LuxCoreSocketColor", "Texture 3", [0.0, 0.0, 0.8])
```
Comment (Theverat) [inline nodes/textures/triplanar.py]: Since these textures correspond to coordinate axis X, Y, Z, maybe we should name them as such so the user knows which texture is on which axis?
By the way, are they aligned to the local (object) or world coordinate system?

Comment (Dade916) [reply nodes/textures/triplanar.py]: I suggested @Alpistinho, in LuxCore pull request, to use a TextureMapping3D (GlobalMapping3D or LocalMapping3D) to map the hit point so the user can manipulate the trilinear texture as he wants.

Comment (Theverat) [reply nodes/textures/triplanar.py]: Ok, so the node is missing a "LuxCoreSocketMapping3D" input.

You can look at another node how to add this, for example here:
https://github.com/LuxCoreRender/BlendLuxCore/blob/7f15ec345c1df215e1389b05bda9ae28663bb608/nodes/textures/blenderclouds.py#L26
And here it's exported:
https://github.com/LuxCoreRender/BlendLuxCore/blob/7f15ec345c1df215e1389b05bda9ae28663bb608/nodes/textures/blenderclouds.py#L42
https://github.com/LuxCoreRender/BlendLuxCore/blob/7f15ec345c1df215e1389b05bda9ae28663bb608/nodes/textures/blenderclouds.py#L53

**Comment by Theverat** [comment]
There are still 3 things to fix for this project on the BlendLuxCore side, see the conversation items above (fix capitalization in menu, remove changes in aovs.py, add uvmapping3d socket).
@Alpistinho in case you don't have time, I can do this stuff in a few days.

**Comment by Alpistinho** [comment]
Sorry, I've got the flu and couldn't look at this earlier.

I've reverted the AOV change and fixed the capitalization (Mapping is uppercased all around)
About the socket, the code is already there, no?

**Comment by Theverat** [comment]
That's unfortunate, I hope you get well soon!
The pull request looks good now, thanks for the changes.
